relay-wstring/src/lib.rs

```diff
@@ -121,7 +121,7 @@ impl WStr {
         if index == 0 || index == self.len() {
             return true;
         }
-        if index % 2 != 0 {
+        if index % 2 != 0 || index > self.len() {
```

Comment:
If you want panic-less code below, you can also do this:

```rust
self.raw.get(index..index+2).map_or(false, |s| !is_trailing_surrogate(u16::copy_from_le_bytes(s))
```

Unfortunately, that does not get rid of the panicking in `copy_from_le_bytes`, but at least from the slice access. It would be phenomenal, if we could get rid of the `assert!` in `copy_from_le_bytes` somehow, but I think this is only possible if you change the signature to return `Option<u16>`:

```rust
self.raw
  .get(index..)
  .and_then(u16::copy_from_le_bytes)
  .map_or(false, |u| !is_trailing_surrogate(u))
```

Overall, I think this would be the cleanest change.